symfony/plugins/orangehrmCorePlugin/Service/MenuService.php

```diff
@@ -68,291 +59,210 @@ public function getMenuDao(): MenuDao
-        $firstLevelHolder = [];
-
-        foreach ($firstLevelItems as $firstLevelItem) {
-            $secondLevelItems = $firstLevelItem->getSubMenuItems();
-            $secondLevelHolder = [];
-
-            if (!empty($secondLevelItems)) {
-                foreach ($secondLevelItems as $secondLevelItem) {
-                    $thirdLevelItems = $secondLevelItem->getSubMenuItems();
-                    $thirdLevelHolder = [];
-
-                    if (!empty($thirdLevelItems)) {
-                        foreach ($thirdLevelItems as $thirdLevelItem) {
-                            $menuItemDetails = $this->_extractMenuItemToArray($thirdLevelItem);
-                            $this->_populateMenuInfoArrays($thirdLevelItem, $menuItemDetails);
-                            $thirdLevelHolder[] = $menuItemDetails;
-                        }
-                    }
-
-                    $menuItemDetails = $this->_extractMenuItemToArray($secondLevelItem);
-                    $menuItemDetails['subMenuItems'] = $thirdLevelHolder;
-                    $this->_populateMenuInfoArrays($secondLevelItem, $menuItemDetails);
-                    $secondLevelHolder[] = $menuItemDetails;
-                }
-            }
-
-            $menuItemDetails = $this->_extractMenuItemToArray($firstLevelItem);
-            $menuItemDetails['subMenuItems'] = $secondLevelHolder;
-            $this->_populateMenuInfoArrays($firstLevelItem, $menuItemDetails);
-            $firstLevelHolder[] = $menuItemDetails;
-        }
-
-        return [
-            'menuItemArray' => $firstLevelHolder,
-            'actionArray' => $this->actionArray,
-            'parentIdArray' => $this->parentIdArray,
-            'levelArray' => $this->levelArray
-        ];
-    }
-
-    /**
-     * @param MenuItem $menuItem
-     * @param array $menuItemDetails
-     */
-    private function _populateMenuInfoArrays(MenuItem $menuItem, array $menuItemDetails): void
-    {
-        $this->parentIdArray[$menuItemDetails['id']] = $menuItem->getParent() instanceof MenuItem
-            ? $menuItem->getParent()->getId() : null;
-        $this->levelArray[$menuItemDetails['id']] = $menuItemDetails['level'];
-
-        if (!empty($menuItemDetails['module']) && !empty($menuItemDetails['action'])) {
-            $this->actionArray[$menuItemDetails['module'] . '_' . $menuItemDetails['action']] = $menuItemDetails['id'];
+        if (!$this->screenPermissionService instanceof ScreenPermissionService) {
+            $this->screenPermissionService = new ScreenPermissionService();
         }
+        return $this->screenPermissionService;
     }
 
     /**
      * @param string $moduleName
      * @param array $menuTitles
      * @return int
-     * @throws DaoException
      */
     public function enableModuleMenuItems(string $moduleName, array $menuTitles = []): int
     {
         return $this->getMenuDao()->enableModuleMenuItems($moduleName, $menuTitles);
     }
```

Comment:
like a wrapper function can it be removed?

Reply:
yes. will try to attempt it. used multiple places

Reply:
this is removed and again added to reset menu cache